### PR TITLE
Added notes on editor_clientserver cvar

### PR DESCRIPTION
**Pull request by AMZN-Olex**
Signed-off-by: Olex Lozitskiy <5432499+AMZN-Olex@users.noreply.github.com>

## Change summary

Added information on the new CVar from Multiplayer gem: editorsv_clientserver.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/networking/settings.md [hunk 1]
```diff
@@ -111,12 +111,13 @@ These settings control how a server will be launched during the "play-in-editor"
...
+| editor_clientserver                | Whether the Editor should act as a server and a client at the same time, without launching a dedicated server process.  | False | Only applies if `editorsv_enabled` is also true. |
+| editorsv_hidden                    | If true the Editor will automatically launch a server upon entering game mode. Set `editorsv_hidden` to true if you want the launched server to be started as a background process without any visible window.                                                                                                           | False     |                                                                                                                                                  |
```
Comment (AMZN-puvvadar) [inline content/docs/user-guide/networking/settings.md:120]:
```suggestion
| editorsv_hidden                    | If the Editor is configured to automatically launch a server upon entering game mode, set `editorsv_hidden` to true if you want the launched server to be started as a background process without any visible window.                                                                                                           | False     |                                                                                                                                                  |
```

This description doesn't make sense to me. You may want to include detail on how editorsv_enabled and edtiro_clientserver matter here. A docs maintainer's opinion here would be great.

File: content/docs/user-guide/networking/settings.md [hunk 2]
```diff
@@ -111,12 +111,13 @@ These settings control how a server will be launched during the "play-in-editor"
...
 | editorsv_enabled                   | If true the Editor will attempt to connect to a Multiplayer server upon entering game mode.                                                                                                                                                                                                                              | False     ||
-| editorsv_hidden                    | If true the Editor will automatically launch a server upon entering game mode. Set `editorsv_hidden` to true if you want the launched server to be started as a background process without any visible window.                                                                                                           | False     |                                                                                                                                                  |  
+| editor_clientserver                | Whether the Editor should act as a server and a client at the same time, without launching a dedicated server process.  | False | Only applies if `editorsv_enabled` is also true. |
```
Comment (AMZN-puvvadar) [inline content/docs/user-guide/networking/settings.md:119]: Should this be editorsv_clientserver?

File: content/docs/user-guide/networking/settings.md (hunk 1 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/networking/settings.md]: I reverted my whitespace changes. I didn't change any of this in this PR.

File: content/docs/user-guide/networking/settings.md (hunk 2 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/networking/settings.md]: Yes, done.

**Review by AMZN-puvvadar — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

**Pull request merged**